artificial-scrum-master/apps/artificial.scrum.master/Artificial.Scrum.Master.ScrumIntegration/Features/Timeline/GetProfileTimeLineService.cs

```diff
@@ -1,14 +1,69 @@
+using Artificial.Scrum.Master.ScrumIntegration.Exceptions;
+using Artificial.Scrum.Master.ScrumIntegration.Infrastructure.ApiTokens;
+using Artificial.Scrum.Master.ScrumIntegration.Infrastructure.ScrumServiceHttpClient;
+using Artificial.Scrum.Master.ScrumIntegration.Utilities;
+
 namespace Artificial.Scrum.Master.ScrumIntegration.Features.Timeline;
 
 internal interface IGetProfileTimeLineService
 {
-    Task<GetProfileTimeLineResponse> Handle(string userId);
+    Task<List<GetProfileTimeLineResponse>> Handle(string userId);
 }
 
 internal class GetProfileTimeLineService : IGetProfileTimeLineService
 {
-    public Task<GetProfileTimeLineResponse> Handle(string userId)
+    private readonly IUserTokensRepository _userTokensRepository;
+    private readonly IProjectHttpClientWrapper _projectHttpClientWrapper;
+    private readonly IJwtDecoder _jwtDecoder;
+
+    public GetProfileTimeLineService(IUserTokensRepository userTokensRepository,
+        IProjectHttpClientWrapper projectHttpClientWrapper, IJwtDecoder jwtDecoder)
+    {
+        _userTokensRepository = userTokensRepository;
+        _projectHttpClientWrapper = projectHttpClientWrapper;
+        _jwtDecoder = jwtDecoder;
+    }
+
+    public async Task<List<GetProfileTimeLineResponse>> Handle(string userId)
```

Comment:
nie podoba mi się to co zwraca ta metoda. Ten model response powinien być modelem który zwraca nasze api, a tutaj robisz z niego listę? I co potem zwracasz w api?

Reply:
Wyobraź sobie taki scenariusz, oprócz listy tych rzeczy chcemy zwrócić jeszcze jakiegoś boola, to co zrobisz w endpoincie? W takiej sytuacji zostanie ci AnnonymousObject chyba tylko albo kolejny typ który i tak owrapuje tego responsa :)

Reply:
I do tego dochodzi jeszcze jeden problem. Teraz w endpoincie zwracasz zawsze liste responseów no i jeżeli byśmy chcieli dodać nową propertę to też powstaną breaking changes bo struktura modelu zmieni się
z: [{}, {}, {}]
na np. {"someBool": false, [{},{},{}]}
No i teraz jeżeli gdzieś robimy zapytanie do tego endpointu no to nam się rozwali kompletnie deserializacja response'a chyba, że wdrożymy zmiany równocześnie
Raczej wolałbym uniknąć czegoś takiego, chociaż wiem, że frontend wdrażamy razem z backendem ale tak na przyszłość to raczej zła praktyka kiedy masz np. komunikaje między mikroserwisami albo api dla kilku frontendów. Taka zmiana wtedy wywala kilka apek i nie da się temu zapobiedz w żaden sposób